ext/contributioncancelactions/contributioncancelactions.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+require_once 'contributioncancelactions.civix.php';
+// phpcs:disable
+use CRM_Contributioncancelactions_ExtensionUtil as E;
+// phpcs:enable
+
+/**
+ * Implements hook_civicrm_preProcess().
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_post
+ */
+function contributioncancelactions_civicrm_post($op, $objectName, $objectId, $objectRef) {
```

Comment:
@eileenmcnaughton We'll need to use civicrm_postCommit (https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_postCommit/) to be safe here.

Reply:
hmm - you could make a case either way - currently it's within the transaction so if it fails the whole thing does - which would make the case that fixing the related entities is integral.